Cannot combine the ARB_VECTORIZE and ARB_ARCH options in CMake

When trying to configure with both these options I am getting a series of the following errors:

```
CMake Error at arbor/CMakeLists.txt:96 (target_link_libraries):
  Error evaluating generator expression:

    $<COMPILE_LANGUAGE:CUDA>

  $<COMPILE_LANGUAGE:...> Unknown language.
```

Here is how to reproduce on Daint:

```
module load daint-mc
module unload PrgEnv-cray
module load PrgEnv-gnu
module load CMake
module unload gcc
module load gcc/7.1.0
export CRAYPE_LINK_TYPE=dynamic
mkdir build
cd build
cmake .. -DARB_WITH_ASSERTIONS=ON -DARB_VECTORIZE=ON -DARB_ARCH=broadwell
make -j
```

--- a/cmake/CompilerOptions.cmake
+++ b/cmake/CompilerOptions.cmake
@@ -135,12 +135,15 @@ function(set_arch_target optvar arch)
         endif()
     endif()
 
-    # Prefix architecture options with `-Xcompiler=` when compiling CUDA sources, i.e.
-    # with nvcc.
-    set(arch_opt_cuda_guarded)
-    foreach(opt ${arch_opt})
-        list(APPEND arch_opt_cuda_guarded "$<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=>${opt}")
-    endforeach()
+    if (ARB_WITH_GPU)
+        # Prefix architecture options with `-Xcompiler=` when compiling CUDA sources, i.e.
+        # with nvcc.
+        set(arch_opt_cuda_guarded)
+        foreach(opt ${arch_opt})
+            list(APPEND arch_opt_cuda_guarded "$<$<COMPILE_LANGUAGE:CUDA>:-Xcompiler=>${opt}")
+        endforeach()
 
-    set("${optvar}" "${arch_opt_cuda_guarded}" PARENT_SCOPE)
+        set("${optvar}" "${arch_opt_cuda_guarded}" PARENT_SCOPE)
+    endif()
+
 endfunction()

